pkg/server/diagnostics/reporter.go

```diff
@@ -94,6 +94,12 @@ type Reporter struct {
 
 	// TestingKnobs is used for internal test controls only.
 	TestingKnobs *TestingKnobs
+
+	// LastSuccessfulTelemetryPing records the current timestamp whenever
+	// we successfully make contact with the registration server. This
+	// timestamp will be updated regardless of whether the response we get
+	// back is successful or not.
+	LastSuccessfulTelemetryPing time.Time
```

Comment:
This may need to be protected with a mutex or AtomicReference. I am pretty sure there will be multiple goroutines that will need to access this.

Reply:
ah thx. I'll push an update.